crates/router/src/connector/nuvei/transformers.rs

```diff
@@ -229,53 +346,230 @@ impl<F, T>
+        if session_token.is_empty() {
+            return Err(errors::ConnectorError::MissingRequiredField {
+                field_name: "session_token",
+            }
+            .into());
+        }
+        let connector_meta: NuveiAuthType = NuveiAuthType::try_from(&item.connector_auth_type)?;
+        let merchant_id = connector_meta.merchant_id;
+        let merchant_site_id = connector_meta.merchant_site_id;
+        let client_request_id = item.attempt_id.clone();
+        let time_stamp = date_time::date_as_yyyymmddhhmmss();
+        let merchant_secret = connector_meta.merchant_secret;
+        let request_data = match item.request.payment_method_data.clone() {
+            api::PaymentMethodData::Card(card) => get_card_info(item, &card),
+            api::PaymentMethodData::Wallet(wallet) => match wallet {
+                api_models::payments::WalletData::GooglePay(gpay_data) => Ok(Self {
+                    payment_option: PaymentOption {
+                        card: Some(Card {
+                            external_token: Some(ExternalToken {
+                                external_token_provider: ExternalTokenProvider::GooglePay,
+                                mobile_token: gpay_data.tokenization_data.token,
+                            }),
+                            ..Default::default()
+                        }),
+                        ..Default::default()
+                    },
+                    ..Default::default()
+                }),
+                api_models::payments::WalletData::ApplePay(apple_data) => Ok(Self {
+                    payment_option: PaymentOption {
+                        card: Some(Card {
+                            external_token: Some(ExternalToken {
+                                external_token_provider: ExternalTokenProvider::ApplePay,
+                                mobile_token: apple_data.payment_data,
+                            }),
+                            ..Default::default()
+                        }),
+                        ..Default::default()
+                    },
+                    ..Default::default()
+                }),
+                api_models::payments::WalletData::PaypalRedirect(_) => Ok(Self {
+                    payment_option: PaymentOption {
+                        alternative_payment_method: Some(AlternativePaymentMethod {
+                            payment_method: AlternativePaymentMethodType::ApmgwExpresscheckout,
+                        }),
+                        ..Default::default()
+                    },
+                    billing_address: Some(BillingAddress {
+                        email: item.request.get_email()?,
+                        country: item.get_billing_country()?,
+                    }),
+                    ..Default::default()
+                }),
+                _ => Err(
+                    errors::ConnectorError::NotImplemented("Payment methods".to_string()).into(),
```

Comment:
There is a `NotSupported` connector error, I think we can use that if this connector does not support a particular payment method